runtime/src/main/java/org/corfudb/runtime/view/Layout.java

```diff
@@ -375,10 +375,10 @@ public IReplicationProtocol getReplicationProtocol(CorfuRuntime r) {
             }
 
             @Override
-            public ClusterStatus getClusterHealthForSegment(LayoutSegment layoutSegment,
-                                                            Set<String> responsiveNodes) {
-                return !responsiveNodes.containsAll(layoutSegment.getAllLogServers())
-                        ? ClusterStatus.UNAVAILABLE : ClusterStatus.STABLE;
+            public ClusterStatus getClusterHealthForSegment(
+                    LayoutSegment layoutSegment, Set<String> responsiveNodes) {
+                return responsiveNodes.containsAll(layoutSegment.getAllLogServers())
```

Comment:
what about the degraded state?

Reply:
it's not possible for a chain replication. If at least one node failed, the status gets UNAVAILABLE